client/state/routing/middleware.js

```diff
@@ -13,6 +13,11 @@ import { ROUTE_SET } from 'state/action-types';
 
 const debug = debugFactory( 'calypso:restore-last-location' );
 const LAST_PATH = 'last_path';
+const ALLOWED_PATHS = /^\/(stats|plans|view|posts|pages|media|types|themes|sharing|people|plugins|domains)/i;
```

Comment:
I'd be a tad more descriptive than `ALLOWED_PATHS` for this symbol since this file will likely be used in the future for more than this use case.